source/views/news/news-list.tsx

```diff
@@ -50,11 +52,59 @@ export const NewsList = (props: Props): JSX.Element => {
 		isLoading,
 	} = useNews(props.source)
 
-	// remove all entries with blank excerpts
-	// remove all entries with a <form from the list
-	let entries = data
-		.filter((entry) => entry.excerpt.trim() !== '')
-		.filter((entry) => !entry.content.includes('<form'))
+	let entries = React.useMemo(() => cleanEntries(data), [data])
+
+	let [filters, setFilters] = React.useState<ListType[]>([])
+
+	let getStoryCategories = React.useCallback((story: StoryType) => {
+		return story.categories.map((c) => trimStoryCateogry(c))
+	}, [])
+
+	React.useEffect(() => {
+		let allCategories = entries.flatMap((story) => getStoryCategories(story))
+
+		if (allCategories.length === 0) {
+			return
+		}
+
+		let categories = [...new Set(allCategories)].sort()
+		let filterCategories = categories.map((c) => {
+			return {title: c}
+		})
+
+		let newsFilters: ListType[] = [
+			{
+				type: 'list',
+				key: 'category',
+				enabled: true,
+				spec: {
+					title: 'Categories',
+					options: filterCategories,
+					selected: filterCategories,
+					mode: 'OR',
+					displayTitle: true,
+				},
+				apply: {key: 'category'},
+			},
+		]
+		setFilters(newsFilters)
+	}, [entries, getStoryCategories])
+
+	let filterStories = React.useCallback(() => {
```

Comment:
Similarly here: can we pull this out into a standalone function that takes three parameters?

I'm open to being told "no" :-)

Reply:
I also feel a little bad about re-filtering the entire list on every render, but I'm sure that we have worse performance pitfalls elsewhere, so I won't complain about it here.

Reply:
I won't use the fact that our news lists are small since every optimization is worth exploring. Is there a quick way we can make that more performant or do we tackle that later?

Reply:
use _.memo with a cache size of 1? As a wrapper around our external filter function? Maybe?

Reply:
Done!